### ISSUE TITLE
[DOCS] can't "follow the recommended guide"

### ISSUE THREAD

**Issue by PhilippRoessner** [Badisi/auth-js#30]
### Documentation can be submitted with pull requests

- [X] I know that I can edit the docs myself but prefer to file this issue instead

### Docs URL

https://badisi.github.io/auth-js/site/documentation/getting-started/mobile

### Description

First I'd like to thank you for this great library! It is exactly what I was waiting for!

I am trying to migrate from a complicated mix of oidc-client-ts for web (angular) and ionic-appauth for our capacitor app to this solution. So far the Web Part works great as a proof of concept.

When I try to login on the android capacitor app, I get the following warning in the console:

```
 [@badisi/auth-js] This application is currently using a non recommended browser plugin.

ⓘ Please follow the recommended guide and use `@badisi/capacitor-browsertab` instead.
```

When I look at the code, it seems like this is not required, is that right?

My Authprovider (Identity Server 4) throws an error when I am using the capacitor android client and I am wondering if this might be connected? I remember from my previous implementations that the url scheme was handled differently on ios and on android. One of them had to use "capacitor://" and the other "localhost://" or something like that.

Currently I set the mobileScheme property on the parameter object for initAuth and the same value for the intent-filter in the androidmanifest.xml. Is that all it takes?

**Comment by Badisi** [comment]
Hi @PhilippRoessner, thanks for the kind words!

This library clearly lacks documentation.. sorry about that.
But I might have some free time next month to update it ;-)

Meanwhile, you can have a look at the [getting-started/providers](https://badisi.github.io/auth-js/site/documentation/getting-started/providers) page, where you have all the necessary settings that needs to be set in your IDP.

For mobile to work, you have to allow this two urls:
> demo-app://localhost/?oidc-callback=login
> demo-app://localhost/?oidc-callback=logout

Where **demo-app** is the `custom url scheme` that needs to be set in:
- the `init` of this library (ie. mobileScheme settings through initAuth)
- the [AndroidManifest.xml](https://github.com/Badisi/auth-js/blob/main/projects/demo-app/capacitor/android/app/src/main/AndroidManifest.xml#L30) file of capacitor android
- the [Info.plist](https://github.com/Badisi/auth-js/blob/main/projects/demo-app/capacitor/ios/App/App/Info.plist#L30) file of capacitor iOS

You will also have to install these packages:
- `@capacitor/browser`, so that you can log in a modal window
- `@capacitor/preferences` (or `capacitor-secure-storage-plugin` - more secure), so that your tokens can be stored locally

Do not hesitate to have a look at the [capacitor demo-app](https://github.com/Badisi/auth-js/tree/main/projects/demo-app/capacitor).

---

The warning is about a work in progress and can be safely ignored right now.
Long story short, the `@capacitor/browser` is using the `SFSafariViewController` implementation on iOS which do not share cookies with Safari (results is that SSO cannot work). So I'm planning to release a Capacitor plugin that implements more recents implementations. You can read more about this in this [article](https://developer.okta.com/blog/2022/01/13/mobile-sso#:~:text=SFSafariViewController%20no%20longer%20shares%20cookies,open%20on%20the%20same%20device.).

**Comment by PhilippRoessner** [comment]
I followed the guidelines so far (for Android) and I compared with what I've found in the demo application. Unfortunately the Mobile implementation is not working yet. The Browser opens and after the login it goes back to my app. But it seems like the app is not picking up the token etc. 

I found this in the log:

```
(index):243 native SecureStoragePlugin.get (#95697034)
(index):250 ObjectcallbackId: "95697034"methodName: "get"options: {key: 'oidc.user:https://login-dev.-----.com:WebApp2'}pluginId: "SecureStoragePlugin"[[Prototype]]: Object
(index):217 result SecureStoragePlugin.get (#95697034)
(index):225 {message: 'Item with given key does not exist'}

```

I am subscribing to a few observables of the AuthService (userSession$, accessToken$) but they never fire. 

The Webapp with Angular is working fine. Do you have an idea what I could be missing?

**Comment by Badisi** [comment]
You can enable the log within the library like this:
```ts
import { initAuth, Log } from '@badisi/ngx-auth';

initAuth({
    logLevel: Log.DEBUG
});
```
Then open a **Chrome** browser and type the following address: `chrome://inspect/#devices`.
Once there you should be able to see your device/simulator in the list and an `inspect` button.

Have a look at those logs (especially the ones from `[MobileWindow]`).
Like `[MobileWindow] navigate: successful response: XXX`.

You can also copy/paste them here so I can have a look.

**Comment by PhilippRoessner** [comment]
Sorry for the bad format, here is the log: (interesting stuff seems to be at the end of the log)

```logtalk
[UserManager] getUser: user not found in storage
/favicon.ico:1          Failed to load resource: the server responded with a status of 404 (Not Found)
/favicon.ico:1          Failed to load resource: the server responded with a status of 404 (Not Found)
main.a24512a315ae2891.js:1 [@badisi/auth-js] This application is currently using a non recommended browser plugin.

ⓘ Please follow the recommended guide and use `@badisi/capacitor-browsertab` instead.
(anonymous) @ main.a24512a315ae2891.js:1
prepare @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
t @ polyfills.b22312040db39e45.js:1
Z @ main.a24512a315ae2891.js:1
signinMobile @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
t @ polyfills.b22312040db39e45.js:1
Z @ main.a24512a315ae2891.js:1
login @ main.a24512a315ae2891.js:1
login @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
n.subscribe.s @ main.a24512a315ae2891.js:1
_next @ main.a24512a315ae2891.js:1
next @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
_next @ main.a24512a315ae2891.js:1
next @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
_next @ main.a24512a315ae2891.js:1
next @ main.a24512a315ae2891.js:1
_subscribe @ main.a24512a315ae2891.js:1
_trySubscribe @ main.a24512a315ae2891.js:1
_trySubscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
_subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
_trySubscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
p @ main.a24512a315ae2891.js:1
h @ main.a24512a315ae2891.js:1
_next @ main.a24512a315ae2891.js:1
next @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
_trySubscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
rC @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
y0 @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
y0 @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
_trySubscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
n.subscribe.s @ main.a24512a315ae2891.js:1
_next @ main.a24512a315ae2891.js:1
next @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
_trySubscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
p @ main.a24512a315ae2891.js:1
h @ main.a24512a315ae2891.js:1
_next @ main.a24512a315ae2891.js:1
next @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
_trySubscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
rC @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
p @ main.a24512a315ae2891.js:1
h @ main.a24512a315ae2891.js:1
_next @ main.a24512a315ae2891.js:1
next @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
_trySubscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
rC @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
p @ main.a24512a315ae2891.js:1
h @ main.a24512a315ae2891.js:1
_next @ main.a24512a315ae2891.js:1
next @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
_complete @ main.a24512a315ae2891.js:1
complete @ main.a24512a315ae2891.js:1
_complete @ main.a24512a315ae2891.js:1
complete @ main.a24512a315ae2891.js:1
_complete @ main.a24512a315ae2891.js:1
complete @ main.a24512a315ae2891.js:1
f @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
_complete @ main.a24512a315ae2891.js:1
complete @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
_trySubscribe @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
w @ main.a24512a315ae2891.js:1
subscribe @ main.a24512a315ae2891.js:1
(index):243 native Browser.addListener (#11860707)
(index):243 native Browser.open (#11860708)
(index):217 result Browser.open (#11860708)
(index):217 result Browser.addListener (#11860707)
main.a24512a315ae2891.js:1 [MobileWindow] navigate: error response: Capacitor browser closed by user
error @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
t @ polyfills.b22312040db39e45.js:1
Z @ main.a24512a315ae2891.js:1
onError @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
returnResult @ (index):761
win.androidBridge.onmessage @ (index):731
(index):243 native Browser.removeListener (#11860709)
(index):243 native Browser.close (#11860710)
(index):217 result Browser.close (#11860710)
main.a24512a315ae2891.js:1 [MobileWindow] close: Error: not implemented
    at returnResult ((index):756:32)
    at win.androidBridge.onmessage ((index):731:21)
error @ main.a24512a315ae2891.js:1
(anonymous) @ main.a24512a315ae2891.js:1
invoke @ polyfills.b22312040db39e45.js:1
run @ polyfills.b22312040db39e45.js:1
(anonymous) @ polyfills.b22312040db39e45.js:1
invokeTask @ polyfills.b22312040db39e45.js:1
runTask @ polyfills.b22312040db39e45.js:1
_ @ polyfills.b22312040db39e45.js:1
Promise.then (async)
q @ polyfills.b22312040db39e45.js:1
R @ polyfills.b22312040db39e45.js:1
scheduleTask @ polyfills.b22312040db39e45.js:1
scheduleTask @ polyfills.b22312040db39e45.js:1
scheduleMicroTask @ polyfills.b22312040db39e45.js:1
ee @ polyfills.b22312040db39e45.js:1
z @ polyfills.b22312040db39e45.js:1
(anonymous) @ polyfills.b22312040db39e45.js:1
(anonymous) @ polyfills.b22312040db39e45.js:1
returnResult @ (index):773
win.androidBridge.onmessage @ (index):731
(index):243 native Browser.close (#11860711)
(index):217 result Browser.close (#11860711)
main.a24512a315ae2891.js:1 [MobileWindow] close: Error: not implemented
    at returnResult ((index):756:32)
    at win.androidBridge.onmessage ((index):731:21)
```

**Comment by Badisi** [comment]
Sorry but those logs are not helping.. (nothing important in there).

Also they are a lot missing. Look at what I have on my side (clearly more info about what's going on in the lib):
<img width="1761" alt="Screen Shot 2023-02-21 at 20 50 25" src="https://user-images.githubusercontent.com/674862/220444380-1b0f89cb-7101-4647-ad10-845cda20b915.png">

Two more things:
1. Have you tried on iOS ?
2. Have you tried running the [demo-app capacitor android](https://github.com/Badisi/auth-js/blob/main/projects/demo-app/capacitor/package.json#L8) from this repo ?

**Comment by PhilippRoessner** [comment]
I gave it another shot with the demo-app and it worked perfectly. Then I began to compare everything and noticed that the package @capacitor/app was missing in my project.

It works now perfectly for android. Big Thanks and again Kudos for this project! This is was my expectation on how to implement oAuth into my hybrid app 3 years ago.

My Macbook is currently not ready for development but my colleague is going to try the ios part in the next few days. I'm sure it should be fine :-)

**Comment by PhilippRoessner** [comment]
I just tried it now with ios (simulator) but unfortunately the redirect is not working. This what I see after I try to login:


![Simulator Screen Shot - iPhone 8 - 2023-02-24 at 08 17 13](https://user-images.githubusercontent.com/6959716/221118721-a1cbd347-022c-441d-a684-b839f53eaa57.png)

**Comment by Badisi** [comment]
Glad you got it working on Android 😉
Regarding your issue on iOS, I can't help with just a screenshot.. Have you tried the demo-app too ?

**Comment by PhilippRoessner** [comment]
I just fixed it now, forgot the url scheme in ios (not enough coffee yet lol)

```
	<key>CFBundleURLTypes</key>
	<array>
	<dict>
		<key>CFBundleURLName</key>
		<string>com.getcapacitor.capacitor</string>
		<key>CFBundleURLSchemes</key>
		<array>
		<string>url scheme goes here</string>
		</array>
	</dict>
```

**Comment by Badisi** [comment]
> Where **demo-app** is the `custom url scheme` that needs to be set in:
> 
> - the `init` of this library (ie. mobileScheme settings through initAuth)
> - the [AndroidManifest.xml](https://github.com/Badisi/auth-js/blob/main/projects/demo-app/capacitor/android/app/src/main/AndroidManifest.xml#L30) file of capacitor android
> - the [Info.plist](https://github.com/Badisi/auth-js/blob/main/projects/demo-app/capacitor/ios/App/App/Info.plist#L30) file of capacitor iOS

Yes it was part of what I told you to do ☕ 😅

**Comment by Badisi** [comment]
Anyway, enjoy the lib now and thanks again for the support 😊

**Comment by mordechai-s** [comment]
Hi,

What's about these errors on android? 
![image](https://github.com/Badisi/auth-js/assets/93929177/93d808b8-2128-4264-b667-6845fb268ce3)

**Comment by Badisi** [comment]
Hello @mordechai-s! Thanks for the kind words 😊

1. "navigate: error response: Capacitor browser closed by user"
This error means that the authentication process did not complete. The reason being that the user decided to manually close the authentication window instead of login in.

2. "close: Error: not implemented"
This error is thrown by Capacitor itself. Capacitor has a `window.close()` api on iOS but not yet implemented on Android.
There is still an opened issue regarding this bug here: https://github.com/ionic-team/capacitor-plugins/issues/50
In the library I'm using this api to programmatically close the authentication window when the authentication completes. But I think that in this scenario it was not an issue on Android as the window was closed by itself.

Are you facing any issue with these errors ?
Maybe I could silent the second one on Android.. (at least until the PR is merged).

**Comment by mordechai-s** [comment]
@Badisi Thanks for your response.

No, I'm not facing any issues with these errors, I just wanted to know what they are.

2 more questions,

1. How does silent renew work on mobile, does it use an iframe, or it uses a refresh token?
2. Does the token get removed from secure storage when the user closes the app? how can I do that the user should get logged out when closing the app?

Thanks again

**Comment by Badisi** [comment]
@mordechai-s 

`Silent renew` is always made with `refresh token` (except when the app starts on desktop where the iFrame is used instead).

### Explanation
**On desktop**
As a good security practise, tokens are not stored on desktop - they are kept in the app memory.
So as long as you are logged in and the app is live, when your access token is going to expired, the refresh token will be used to renew the user access. But once the app is closed, the tokens are lost.
Reopening the app (or simply refreshing the page), means that the user will not be re-logged in automatically (even if his session is still opened).
So in case the lib was configured with `retrieveUserSession: true` -> an hidden iFrame will be used to contact the IDP and retrieve the tokens when the app starts.

**On mobile**
Tokens are stored in the device so that the user could be automatically re-logged in when reopening the app.
Depending on the plugins available in your app, the storage will be (by priority order):
1. capacitor-secure-storage-plugin
2. @capacitor/preferences
3. @capacitor/storage
4. localStorage

-----

Regarding your question 2.,
a. If your concern is getting sure that the tokens are removed then it is a matter of knowing when the app is getting closed and removing them (maybe using a `destroy` handler available in your dev framework or listening to an `appClose` event somewhere ?). But I'm not aware of such a feature in Capacitor so I'm not sure if that would be possible.
b. If you concern is just making sure that the user would not be relogged in automatically then you can simply set `retrieveUserSession` to `false`.

**Comment by mordechai-s** [comment]
@Badisi 

Sorry, but I'm not clear on the silent renew, on regular web app we don't need the 'offline_access' scope as the renew works through a hidden iframe, what's about mobile, do I need to add the 'offline_access' scope (and enable it on the IDP level)?

**Comment by ms-emp** [comment]
@Badisi 

I see that the library is using the system browser for login, is there a reason why you don't use an in-app browser like Custom Tabs on Android for a better user experience?

**Comment by Badisi** [comment]
This library will never use the "system browser" as it is not considered a **best practice**.
It is not **secured** and not even **accepted** by Apple while doing authentication.

Right now the library is expecting [`@capacitor/browser`](https://capacitorjs.com/docs/apis/browser) (which is an "in-app browser") to be installed. But I'm also planning to develop another library (later on) which will provide Custom Tabs browsers.

**Comment by ms-emp** [comment]
@Badisi 

Does silent renew run when the app is in the background?

**Comment by Badisi** [comment]
@ms-emp, no, silent renew is not running while the app is in background.

This is an OS limitation, especially on iOS where `setTimeout` and `setInterval` (which are used for silent renew) are paused.
You can find more detailed explanation here: https://capacitorjs.com/docs/v2/apis/background-task#background-task-guidelines

**Comment by ms-emp** [comment]
@Badisi Thanks for your response,

So do you have any idea how I can implement to retrieve the user's session when the user reopens the app without redirecting to login? it looks like `retrieveUserSession` is only being used on desktop (see [here](https://github.com/Badisi/auth-js/blob/66712331c68adec14d553b125576e23c1acbe99c/projects/auth-js/oidc/oidc-auth-manager.ts#L165))

Thanks again for your help
